core/tauri/src/menu/check.rs

```diff
@@ -0,0 +1,122 @@
+// Copyright 2019-2023 Tauri Programme within The Commons Conservancy
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-License-Identifier: MIT
+
+use crate::{run_main_thread, AppHandle, Manager, Runtime};
+
+/// A menu item inside a [`Menu`] or [`Submenu`] and contains only text.
+///
+/// [`Menu`]: super::Menu
+/// [`Submenu`]: super::Submenu
+pub struct CheckMenuItem<R: Runtime> {
+  pub(crate) id: u32,
+  pub(crate) inner: muda::CheckMenuItem,
+  pub(crate) app_handle: AppHandle<R>,
+}
+
+impl<R: Runtime> Clone for CheckMenuItem<R> {
+  fn clone(&self) -> Self {
+    Self {
+      id: self.id,
+      inner: self.inner.clone(),
+      app_handle: self.app_handle.clone(),
+    }
+  }
+}
+
+/// # Safety
+///
+/// We make sure it always runs on the main thread.
+unsafe impl<R: Runtime> Sync for CheckMenuItem<R> {}
+unsafe impl<R: Runtime> Send for CheckMenuItem<R> {}
+
+impl<R: Runtime> super::sealed::IsMenuItemBase for CheckMenuItem<R> {
+  fn inner(&self) -> &dyn muda::IsMenuItem {
+    &self.inner
+  }
+}
+
+impl<R: Runtime> super::IsMenuItem<R> for CheckMenuItem<R> {
+  fn kind(&self) -> super::MenuItemKind<R> {
+    super::MenuItemKind::Check(self.clone())
+  }
+
+  fn id(&self) -> u32 {
+    self.id
+  }
+}
+
+impl<R: Runtime> CheckMenuItem<R> {
+  /// Create a new menu item.
+  ///
+  /// - `text` could optionally contain an `&` before a character to assign this character as the mnemonic
+  /// for this menu item. To display a `&` without assigning a mnemenonic, use `&&`.
+  pub fn new<M: Manager<R>, S: AsRef<str>>(
```

Comment:
I think we should force users to use the Builder and keep this private. It'll be easier to manage changes in the future.

Reply:
I honestly think we should keep both and let users decided, due to formatting, sometimes it maybe easier to just use this method instead of the builder.